File: Depvar/sfran_loglike.r
# sfran_loglike.r
sfran_loglike<-function(mu,sigma2_mu,sigma2,y){
  d = dim(y)[1];
  ni = dim(y)[2];
  Sigmai = sigma2*diag(ni) + sigma2_mu*matrix(rep(1,ni*ni),nrow=ni);
  l = -(ni*d)/2*log(2*pi) - d/2*log(det(Sigmai));
  for (i in 1:d){
  yi = matrix(y[i,]);
  l = l - .5*t(yi-mu) %*% solve(Sigmai,(yi-mu));    
  }
  return (l);
}

File: Depvar/sfran_loglike.jl
function sfran_loglike(mu,sigma2_mu,sigma2,y)
   d, ni = size(y) 
   Sigmai = sigma2*diagm(ones(ni)) .+ sigma2_mu*ones(ni,ni)
   l = -(ni*d)/2*log(2*pi) - d/2*log(det(Sigmai))
   for i=1:d
       yi = y[i,:];
       l = l - .5*(yi .- mu)'*(Sigmai\(yi .- mu))
   end
   return l
end
